gateway-js/src/loadCsdlFromStorage.ts

```diff
@@ -0,0 +1,133 @@
+import { fetch } from 'apollo-server-env';
+import { GraphQLError } from 'graphql';
+
+export const CSDL_QUERY = `#graphql
+  query Csdl($apiKey: String!, $ref: String!) {
+    routerConfig(
+      ref: $ref
+      apiKey: $apiKey
+    ) {
+      __typename
+      ... on RouterConfigResult {
+        id
+        csdl
+      }
+      ... on FetchError {
+        code
+        message
+      }
+    }
+  }
+`;
+
+type CsdlQueryResult = CsdlQuerySuccess | CsdlQueryFailure;
+
+interface CsdlQuerySuccess {
+  data: { routerConfig: RouterConfigResponse };
+}
+
+interface CsdlQueryFailure {
+  data?: { routerConfig: RouterConfigResponse };
+  errors: GraphQLError[];
+}
+
+type RouterConfigResponse = RouterConfigResult | FetchError;
+
+interface RouterConfigResult {
+  __typename: 'RouterConfigResult';
+  id: string;
+  csdl: string;
+  messages: Message[];
+}
+
+interface Message {
+  __typename: 'Message';
+  body: string;
+  level: MessageLevel;
+}
+
+enum MessageLevel {
+  ERROR = 'ERROR',
+  WARN = 'WARN',
+  INFO = 'INFO',
+}
+
+interface FetchError {
+  __typename: 'FetchError';
+  code: FetchErrorCode;
+  message: string;
+}
+
+enum FetchErrorCode {
+  AUTHENTICATION_FAILED = 'AUTHENTICATION_FAILED',
+  ACCESS_DENIED = 'ACCESS_DENIED',
+  UNKNOWN_REF = 'UNKNOWN_REF',
+  RETRY_LATER = 'RETRY_LATER',
+}
+
+export async function loadCsdlFromStorage({
+  graphId,
+  graphVariant,
+  apiKey,
+  fetcher,
+}: {
+  graphId: string;
+  graphVariant: string;
+  apiKey: string;
+  fetcher: typeof fetch;
+}) {
+  const result = await fetcher(
+    'https://us-central1-mdg-services.cloudfunctions.net:443/cloudconfig-staging/',
+    {
+      method: 'POST',
+      body: JSON.stringify({
+        query: CSDL_QUERY,
+        variables: {
+          ref: `${graphId}@${graphVariant}`,
+          apiKey,
+        },
+      }),
+      headers: {
```

Comment:
I think we may want to include the user agent content or something similar as client name/version headers for metrics tagging?

Reply:
I didn't address this yet and now I'm not sure what you mean. Are you asking for something beyond what I have in the `user-agent` header now? Can you clarify?

Reply:
Resolved in 3aa52339